src/main/java/org/sqlite/hibernate/dialect/SQLiteDialect.java

```diff
@@ -327,9 +331,55 @@ public UniqueDelegate getUniqueDelegate() {
 		private SQLiteUniqueDelegate(Dialect dialect) {
 			super( dialect );
 		}
+
+		/**
+		 * SQLite use table creation sql to define unique constraints.
+		 */
 		@Override
 		public String getColumnDefinitionUniquenessFragment(Column column) {
-			return " unique";
+			return "";
+		}
+
+		/**
+		 * SQLite uses table creation sql to define unique constraints, and do not support alter table sql to add
+		 * constraints.
+		 * Such as "create table person( first_name varchar(255),last_name varchar(255),unique(first_name, last_name) )".
+		 */
+		@Override
+		public String getTableCreationUniqueConstraintsFragment(Table table) {
+			// get all uniqueKeys
+			StringBuilder builder = new StringBuilder();
+			Iterator<UniqueKey> iter = table.getUniqueKeyIterator();
+			while(iter.hasNext()) {
+				builder.append(", unique(");
+				UniqueKey key = iter.next();
+				Iterator<Column> columnIter = key.getColumnIterator();
+				while(columnIter.hasNext()) {
+					Column column = columnIter.next();
+					builder.append(column.getName());
+					if (columnIter.hasNext()) {
+						builder.append(",");
+					}
+				}
+				builder.append(")");
```

Comment:
Why not reuse `org.hibernate.dialect.unique.DefaultUniqueDelegate#uniqueConstraintSql` ?

Reply:
Yes, it is better to use the DefaultUniqueDelegate#uniqueConstraintSql() function, I did not noticed it.
I will change and test it.